pkg/influxid/id.go

```diff
@@ -0,0 +1,123 @@
+package influxid
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"strconv"
+)
+
+// IDLength is the exact length a string (or a byte slice representing it) must have in order to be decoded into a valid ID.
+const IDLength = 16
+
+var (
+	// ErrInvalidID signifies invalid IDs.
+	ErrInvalidID = errors.New("invalid ID")
+
+	// ErrInvalidIDLength is returned when an ID has the incorrect number of bytes.
+	ErrInvalidIDLength = errors.New("id must have a length of 16 bytes")
+)
+
+// ID is a unique identifier.
+//
+// Its zero value is not a valid ID.
+type ID uint64
```

Comment:
Higher-level question: does the CLI need to care about the difference between and ID and an arbitrary `string`? I was thinking about it while porting `influx bucket`, and had these assumptions:
* The server-side will always have its own validation for IDs
* The CLI will only ever be passing string IDs from CLI args -> server and string IDs from server -> stdout

Is it better for the CLI to be agnostic to what makes a "valid" ID as a future-proofing mechanism? Or better to push the error message as close to the original data entry as possible?

Reply:
I would advocate it is better to do some parsing if the format is known, to fail faster. No different to parsing an integer flag into an `int` arg, even if it is sent as a string to the server. If our ID format changes, we can amend the ID parser.